presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -335,6 +338,21 @@ protected Boolean visitFunctionCall(FunctionCall node, Void context)
                             PERFORMANCE_WARNING,
                             "COUNT(DISTINCT xxx) can be a very expensive operation when the cardinality is high for xxx. In most scenarios, using approx_distinct instead would be enough"));
                 }
+                if (functionResolution.isApproxDistinctFunction(analysis.getFunctionHandle(node))) {
+                    double maxStandardError = LOWEST_APPROX_DISTINCT_MAX_STANDARD_ERROR;
+                    // maxStandardError is supplied
+                    if (node.getArguments().size() > 1) {
+                        Expression maxStandardErrorExpr = node.getArguments().get(1);
+                        if (maxStandardErrorExpr instanceof DoubleLiteral) {
+                            maxStandardError = ((DoubleLiteral) maxStandardErrorExpr).getValue();
+                        }
+                    }
+                    if (maxStandardError <= LOWEST_APPROX_DISTINCT_MAX_STANDARD_ERROR) {
+                        warningCollector.add(new PrestoWarning(
+                                PERFORMANCE_WARNING,
+                                String.format("approx_distinct can be a very expensive operation when the max standard error is too low (<=%f)", LOWEST_APPROX_DISTINCT_MAX_STANDARD_ERROR)));
```

Comment:
The warning message is more like "approx_distinct can produce low-precision result with the existing standard error %s". Something like that.